crates/bevy_gizmos/src/gizmos.rs

```diff
@@ -589,6 +628,59 @@ impl Drop for Circle2dBuilder<'_, '_> {
     }
 }
 
+/// A builder returned by [`Gizmos::arc_2d`].
+/// `direction_angle` is the angle where the arc will be centered. `0.0`
+/// `arc_angle` is defining the amount of the underlying circle
+/// being drawn. With the `radius`, it will define the length of the arc.
+pub struct Arc2dBuilder<'a, 's> {
+    gizmos: &'a mut Gizmos<'s>,
+    position: Vec2,
+    direction_angle: f32,
+    arc_angle: f32,
+    radius: f32,
+    color: Color,
+    segments: Option<usize>,
+}
+
+impl Arc2dBuilder<'_, '_> {
+    /// Set the number of line-segements for this arc.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.segments = Some(segments);
+        self
+    }
+}
+
+impl Drop for Arc2dBuilder<'_, '_> {
+    fn drop(&mut self) {
+        let segments = match self.segments {
+            Some(segments) => segments,
+
+            // Do a linear interpolation between 1 and `DEFAULT_CIRCLE_SEGMENTS`
+            // using the arc angle as scalar.
+            None => (1. + (self.arc_angle.abs() / TAU) * (DEFAULT_CIRCLE_SEGMENTS as f32 - 1.))
+                .round() as usize,
```

Comment:
```suggestion
            None => ((self.arc_angle.abs() / TAU) * DEFAULT_CIRCLE_SEGMENTS).ceil() as usize,
```
It seems correct without adding and subtracting one, and using `ceil` to avoid rounding down to zero when the arc is smaller than `TAU / 32 / 2`